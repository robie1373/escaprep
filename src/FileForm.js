import React, { Component } from 'react';
import Papa from 'papaparse';
import ReactFileReader from 'react-file-reader';
import UnorderedList from './UnorderedList';

class FileForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
      inFile: ''
    };

    this.parseFiles = this.parseFiles.bind(this);
    this.testableReactFileReader = this.testableReactFileReader.bind(this);
    //this.stateSetter = this.stateSetter.bind(this);
  }

  componentDidCatch(error, info) {
      // Display fallback UI
      this.setState({ hasError: true });
      // You can also log the error to an error reporting service
      console.log(error);
      console.log(info);
    }
/*
  stateSetter = data => {
    this.props.stateSetter(data);
  }
*/
  parseFiles = files => {
    Papa.parse(files[0], {
      header: true,
      complete: (results) => {
        //this.setState({parsedData: results})
        this.props.stateSetter(results);
      }
    });
  }

  testableReactFileReader = () => {
    if (this.props.filereader) {
      return(
        <this.props.filereader />
      );
    }
    else {
      return(
        <ReactFileReader fileTypes={'.csv'} handleFiles={this.parseFiles}>
          <button className='fileUploadButton'>Upload File</button>
        </ReactFileReader>
      );
    }
  }
//  <UnorderedList items={this.props.parsedData} />
  render() {
    if (this.state.hasError) {
      return (
        <div className="has-error">
          <h1> An Error has occurred!</h1>
          <p> See console for details</p>
        </div>
      )
    }
    return (
      <div className="fileForm">

        <this.testableReactFileReader />

        <pre>
          {this.state.inFile}
        </pre>
        <pre>

        </pre>
      </div>
    );
  }
}

export default FileForm;
