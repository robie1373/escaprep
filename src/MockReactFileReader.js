import React, { Component } from 'react';

class MockReactFileReader extends Component {
  render () {
    return (
      <div
         className="react-file-reader"
       >
         <input
           accept=".csv"
           className="react-file-reader-input"
           id="0d8f72bc-f98a-415e-aba9-c2b77dfc243c"
           multiple={false}
           onChange={[Function]}
           type="file"
         />
         <div
           className="react-file-reader-button"
           onClick={[Function]}
         >
           <button
             className="fileUploadButton"
           >
             Upload File
           </button>
         </div>
       </div>
    );
  }
}

export default MockReactFileReader;
