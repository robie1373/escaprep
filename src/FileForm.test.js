import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FileForm from './FileForm';
import MockReactFileReader from './MockReactFileReader';

test('snapshot', () => {
  const component = renderer.create(<FileForm filereader={MockReactFileReader }/>);

  let tree = component.toJSON()
  ;
  expect(tree).toMatchSnapshot();
});
