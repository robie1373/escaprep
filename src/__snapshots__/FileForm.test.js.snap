// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="fileForm"
>
  <div
    className="react-file-reader"
  >
    <input
      accept=".csv"
      className="react-file-reader-input"
      id="0d8f72bc-f98a-415e-aba9-c2b77dfc243c"
      multiple={false}
      onChange={
        Array [
          [Function],
        ]
      }
      type="file"
    />
    <div
      className="react-file-reader-button"
      onClick={
        Array [
          [Function],
        ]
      }
    >
      <button
        className="fileUploadButton"
      >
        Upload File
      </button>
    </div>
  </div>
  <pre>
    
  </pre>
  <pre />
</div>
`;
